config/samples/pipeline_v1beta1_pipeline.yaml

```diff
@@ -7,46 +7,60 @@ spec:
     tasks:
         - name: 'test'
           taskRef:
-              name: test-wizzbang-task
-          sourceBindings:
-              - inputName: 'repoUnderTest'
-                sourceKey: 'wizzbang'
-          nextTasks: ['buildPush']
+            name: test-wizzbang-task
+          inputsourceBindings:
+            - name: repoUnderTest
+              sourceKey: wizzbang
         - name: 'buildPush'
           taskRef:
-              name: build-push-task
-          sourceBindings:
-              - inputName: 'workspace'
-                sourceKey: 'wizzbang'
-          artifactStoreBindings:
-              - storeName: 'registry'
-                storeKey: 'stagingRegistry'
+            name: build-push-task
+          inputSourceBindings:
+            - name: 'workspace'
+              sourceKey: 'wizzbang'
+              passedConstraint: [test-wizzbang-task]
+          outputSourceBindings:
+            - name: 'registry'
+              sourceKey: 'stagingRegistry'
           params:
-              - name: 'pathToDockerfile'
-                value: 'build/Dockerfile'
+            - name: 'pathToDockerfile'
+              type: 'build/Dockerfile'
         - name: 'integrationTest'
           taskRef:
-              name: integration-test-wizzbang-task
-          sourceBindings:
-              - inputName: 'repoUnderTest'
-                sourceKey: 'wizzbang'
-          paramBindings:
-              - inputName: testImage
-                taskName: buildPush
-                taskOutputName: builtImage
-        - name: 'deploy'
+            name: integration-test-wizzbang-task
+          inputSourceBindings:
+            - name: repoUnderTest
+              sourceKey: wizzbang
+              passedConstraint: [test]
+          outputSourceBindings:
+            - name: stagingRegistry
+              sourceKey: stagingRegistry
+              passedConstraint: [build-push-task]
+        - name: deploy
           taskRef:
               name: deploy-with-helm
           params:
-              - name: pathToHelmCharts
-                value: deploy/helm
-          sourceBindings:
-              - inputName: 'repoToDeploy'
-                sourceKey: 'wizzbang'
+            - name: pathToHelmCharts
+              type: deploy/helm
+          inputSourceBindings:
+            - name: repoToDeploy
+              sourceKey: wizzbang
+              passedConstraint: [integration-test-wizzbang-task]
           clusterBindings:
-              - clusterName: prod
-          paramBindings:
-              - inputName: image
-                taskName: buildPush
-                taskOutputName: builtImage
-          prevTasks: ['integrationTest']
+            - clusterName: prod
+          outputSourceBindings:
+            - name: stagingRegistry
+              sourceKey: stagingRegistry
+              passedConstraint: [build-push-task]
+    resources:
```

Comment:
what are your thoughts on declaring these resources in the Pipeline vs the PipelineParams?

I feel like they should be in the PipelineParams - or if not, that we should move everything that was in PipelineParams into Pipeline, but it seems like maybe we have different ideas about what makes sense in each?

Reply:
My preference for using the `Pipeline` is that the `pipelineParams` is like a generic bucket that had items not related to each other. I was going to propose moving all of the types currently in `pipelineParams` to `Pipeline` under their own heading. This way the yaml file is more clear to read and understand

Reply:
Yes i like the idea of separate yaml for resources. Allows you to isolate the different environments